keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,808 @@
+-->
+
+Today only deployment controller has [status](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status) to reflect the state during it's lifecycle. This enhancement extends the scope of 
+those of conditions to other controllers(DaemonSet and StatefulSet).
+
+### Goals
+
+The current status of a workload can be depicted via its conditions. It can be
+- Progressing
+- Complete
+- Failed to Progress
+
+All workload controllers should have above conditions to reflect their states
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Modifying the existing states of deployment controller
+- Changing the definition of States
+- Introduce new api for existing conditions
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As an end-user of Kubernetes, I'd like all my workload controllers to have consistent states
+
+#### Story 2
+As an developer building Kubernetes Operators, I'd like all my operators deployed to have
+consistent states.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+We are proposing 3 new conditions called
+- Progressing
+- Complete
+- Failed
+to be added to DaemonSet and Stateful controllers. The definitions are similar to what we have for [Deployment controller](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status).
+
+#### Progressing
+Kubernetes marks a DaemonSet or Stateful as `progressing` when:
+- The DaemonSet or StatefulSet is created
+- The DaemonSet or StatefulSet is scaling up or scaling down
+- New DaemonSet or StatefulSet pods become Ready or available
+
+#### Complete
+Kubernetes marks a DaemonSet or Stateful as complete when it has the following characteristics:
+
+- All of the replicas associated with the DaemonSet or StatefulSet have been updated to the latest version you've specified, meaning any updates you've requested have been completed.
+- All of the replicas associated with the DaemonSet or StatefulSet are available.
+- No old replicas for the DaemonSet or Stateful are running.
```

Comment:
is the behaviour similar to Deployment's `Available`? What happens when scaling down?